### PR TITLE
Move SwiftASTContext & TypeSystemSwift* into Plugins/TypeSystem/Swift

### PR DESCRIPTION
**Pull request by adrian-prantl**
Move `SwiftASTContext` & `TypeSystemSwift*` into `Plugins/TypeSystem/Swift` to mirror a similar move spearheaded by the Clang typesystem.

### CODE REVIEW

**Comment by adrian-prantl** [comment]
@swift-ci test

**Comment by adrian-prantl** [comment]
Apologies to everyone whose pull requests this is going to break, but with 5.3 slowly winding down, this seems to be the right moment for this long overdue refactoring.

File: lldb/source/Plugins/TypeSystem/Swift/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  TypeSystemSwift.cpp
+  TypeSystemSwiftTypeRef.cpp
+  SwiftASTContext.cpp
+
+  LINK_LIBS
+    lldbCore
+    lldbSymbol
+    lldbTarget
+    lldbUtility
+#    swiftAST
```
Comment (vedantk) [inline lldb/source/Plugins/TypeSystem/Swift/CMakeLists.txt]: Can we go ahead and delete these?

**Review by vedantk — COMMENTED** [review]
Would it be possible to split the pure code-movement and getTypeInfo changes into separate PRs? Or perhaps land the code-movement changes up front?

File: lldb/source/Plugins/TypeSystem/Swift/CMakeLists.txt (hunk 1 above)
Comment (adrian-prantl) [reply lldb/source/Plugins/TypeSystem/Swift/CMakeLists.txt]: > Would it be possible to split the pure code-movement and getTypeInfo changes into separate PRs?

Sure! I already landed that patch with a separate PR, I just didn't want to wait another 3 hours before putting this one up and it was a dependency.

**Comment by adrian-prantl** [comment]
@swift-ci test

**Comment by adrian-prantl** [comment]
@JDevlieghere This is going to master and master/next. Should I cherry-pick it to master-rebranch, too?

**Comment by JDevlieghere** [comment]
> @JDevlieghere This is going to master and master/next. Should I cherry-pick it to master-rebranch, too?

Yes please :-)

**Pull request merged**